api/types/constants.go

```diff
@@ -338,6 +338,13 @@ const (
 	// KindHeadlessAuthentication is a headless authentication resource.
 	KindHeadlessAuthentication = "headless_authentication"
 
+	// KindMaintenanceWindow determines maintenance times for the cluster.
+	KindMaintenanceWindow = "maintenance_window"
```

Comment:
Are we using _maintenance window_ and _upgrade window_ interchangeably or are they separate things? If they are the same thing, I think it would be best to use the same name consistently.

Reply:
Naming is hard. The resource is called "maintenance window".  The only currently supported block is `agent_upgrades`.  Ex:

```yaml
kind: maintenance_window
spec:
  agent_upgrades:
    utc_start_hour: 3
    weekdays:
    - Mon
    - Wed
    - Fri
```

The idea, however, is to allow _other_ maintenance window sub-fields to be added in the future, so there is an "agent upgrade maintenance window" that exists now, but we might add other maintenance window options in the future which might not cover the same time range or purpose, but would still be part of the "maintenance window" config object.  I was originally going to name the top-level resource `ClusterSchedule`, but that felt _too_ generic.

Reply:
If the resource is called _maintenance window_, why cant we use that name in the protos instead of "upgrade window"?